src/js/webrequest.js

```diff
@@ -776,6 +784,16 @@ function dispatcher(request, sender, sendResponse) {
     badger.storage.getBadgerStorageObject("action_map").deleteItem(request.origin);
 
     sendResponse();
+  } else if (request.type == "saveErrorText") {
+    let activeTab = badger.tabData[request.tabId];
```

Comment:
Could `badger.tabData[request.tabId]` ever be missing at this point?

Reply:
tNo, the message comes from popup.js with the tabId set to `POPUP_DATA.tabId` attribute set at the beginning of it's execution.

```
$(function () {
  getTab(function (tab) {
    chrome.runtime.sendMessage({
      type: "getPopupData",
      tabId: tab.id,
      tabUrl: tab.url
    }, (response) => {
      setPopupData(response);
      refreshPopup();
      init();
    });
  });
});
```

I can only think of getTab failing and thus the `tab` variable not being passed in the callback, but that should be taken care of way before the popup messaging logic starts.